packages/askui-nodejs/src/execution/dsl.spec.ts

```diff
@@ -17,63 +16,131 @@ describe('DSL', () => {
           imageCompareFormat: 'grayscale',
           name: 'custom element 2',
         })
         .exec();
       expect(testCommandSpy).toHaveBeenCalledWith(
         'Click on custom element button custom element',
-        [{
-          customImage: 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
-          imageCompareFormat: 'grayscale',
-          name: 'custom element 1',
+        {
+          aiElementNames: [],
+          customElementsJson: [
+            {
+              customImage:
+                'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
+              imageCompareFormat: 'grayscale',
+              name: 'custom element 1',
+            },
+            {
+              customImage:
+                'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
+              imageCompareFormat: 'grayscale',
+              name: 'custom element 2',
+            },
+          ],
         },
+      );
+    });
+
+    test('should call exec function with one ai element', async () => {
+      const underTest = new TestCommand();
+      const testCommandSpy = jest.spyOn(underTest, 'fluentCommandExecutor');
+
+      await underTest
+        .click()
+        .aiElement('ai element')
+        .below()
+        .button()
+        .rightOf()
+        .customElement({
+          customImage:
+            'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
+          name: 'custom element',
+        })
+        .exec();
+      expect(testCommandSpy).toHaveBeenCalledWith(
+        'Click on ai element with name <|string|>ai element<|string|> index 0 below button index 0 right of custom element',
         {
-          customImage: 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
-          imageCompareFormat: 'grayscale',
-          name: 'custom element 2',
+          aiElementNames: ['ai element'],
+          customElementsJson: [
+            {
+              customImage:
+                'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
+              name: 'custom element',
+            },
+          ],
+        },
+      );
+    });
+
+    test('should call exec function with two ai element', async () => {
+      const underTest = new TestCommand();
+      const testCommandSpy = jest.spyOn(underTest, 'fluentCommandExecutor');
+
+      await underTest
+        .click()
+        .aiElement('ai element')
+        .below()
+        .button()
+        .rightOf()
+        .aiElement('ai element 2')
+        .exec();
+      expect(testCommandSpy).toHaveBeenCalledWith(
+        'Click on ai element with name <|string|>ai element<|string|> index 0 below button index 0 right of ai element with name <|string|>ai element 2<|string|>',
```

Comment:
```suggestion
        'Click on ai element with name <|string|> ai element <|string|> index 0 below button index 0 right of ai element with name <|string|> ai element 2 <|string|>',
```

Do these missing spaces matter?